### PR TITLE
[6.3] Katello host tools (BZ:1463809)

### PR DESCRIPTION
**Pull request by ldjebran**
cover: https://bugzilla.redhat.com/show_bug.cgi?id=1463809

```shell
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_host.py::KatelloHostToolsTestCase
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 3 items 
2017-11-08 15:56:17 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_erratum_applicability <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_package_applicability <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_report_package_installed_removed <- robottelo/decorators/__init__.py PASSED

============================================== 3 passed in 741.35 seconds ==============================================
```

### CODE REVIEW

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -2607,6 +2610,212 @@ def test_positive_register_host_ak_with_host_collection(self):
...
+        self.client.run('yum install -y {0}'.format(FAKE_0_CUSTOM_PACKAGE))
+        result = self.client.run(
+            'rpm -q {0}'.format(FAKE_0_CUSTOM_PACKAGE)
+        )
+        self.assertEqual(result.return_code, 0)
+        installed_packages = Host.package_list({
+            'host-id': self.host_info['id'],
+            'search': 'name={0}'.format(FAKE_0_CUSTOM_PACKAGE_NAME)
+        })
+        self.assertIn(
```
Comment (oshtaier) [inline tests/foreman/cli/test_host.py]: please check that `self.assertEqual(len(installed_packages), 1)` or greater than 1, not sure here, but you said always verify len, before check list content, so now I am asking you to do the same =)

**Comment by oshtaier** [comment]
Not sure whether it is correct comment, but can we check that we have limited things that we can do with that tool. I meant that we can install packages or erratas, but nothing more that we can achieve with gofer

**Comment by abalakh** [comment]
@oshtaier +1. Checking that tool doesn't have some 'extra' functionality sounds weird at the first glance, but come to think of it - this way we'll ensure gofer is definitely absent on hosts like use case requires.

**Comment by ldjebran** [comment]
@oshtaier comment addressed
```
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_host.py::KatelloHostToolsTestCase
=================================================================================================== test session starts ====================================================================================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/2.7.13/envs/sat-6.3.0/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1, fauxfactory-1.0.1
collected 4 items                                                                                                                                                                                                           
2017-11-21 15:21:39 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_negative_install_package PASSED
tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_erratum_applicability <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_package_applicability <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloHostToolsTestCase::test_positive_report_package_installed_removed <- robottelo/decorators/__init__.py PASSED

================================================================================================ 4 passed in 927.95 seconds ================================================================================================
```

**Pull request merged**